Helm/opensearch-dashboards/templates/deployment.yaml

```diff
@@ -0,0 +1,156 @@
+        {{- range $key, $value := .Values.labels }}
+        {{ $key }}: {{ $value | quote }}
+        {{- end }}
+      annotations:
+        {{- range $key, $value := .Values.podAnnotations }}
+        {{ $key }}: {{ $value | quote }}
+        {{- end }}
+        {{/* This forces a restart if the configmap has changed */}}
+        {{- if .Values.config }}
+        configchecksum: {{ include (print .Template.BasePath "/configmap.yaml") . | sha256sum | trunc 63 }}
+        {{- end }}
+    spec:
+{{- if .Values.priorityClassName }}
+      priorityClassName: {{ .Values.priorityClassName }}
+{{- end }}
+      securityContext:
+{{ toYaml .Values.podSecurityContext | indent 8 }}
+      serviceAccountName: {{ template "opensearch-dashboards.serviceAccountName" . }}
+      {{- if .Values.hostAliases }}
+      hostAliases: {{ toYaml .Values.hostAliases | nindent 6 }}
+      {{- end }}
+      volumes:
+        {{- range .Values.secretMounts }}
+        - name: {{ .name }}
+          secret:
+            secretName: {{ .secretName }}
+        {{- end }}
+        {{- if .Values.config }}
+        - name: config
+          configMap:
+            name: {{ template "opensearch-dashboards.fullname" . }}-config
+        {{- end }}
+    {{- with .Values.nodeSelector }}
+      nodeSelector:
+{{ toYaml . | indent 8 }}
+    {{- end }}
+    {{- with .Values.affinity }}
+      affinity:
+{{ toYaml . | indent 8 }}
+    {{- end }}
+    {{- with .Values.tolerations }}
+      tolerations:
+{{ toYaml . | indent 8 }}
+    {{- end }}
+    {{- if .Values.imagePullSecrets }}
+      imagePullSecrets:
+{{ toYaml .Values.imagePullSecrets | indent 8 }}
+    {{- end }}
+      {{- if .Values.extraInitContainers }}
+      # Currently some extra blocks accept strings
+      # to continue with backwards compatibility this is being kept
+      # whilst also allowing for yaml to be specified too.
+      initContainers:
+      {{- if eq "string" (printf "%T" .Values.extraInitContainers) }}
+{{ tpl .Values.extraInitContainers . | indent 6 }}
+      {{- else }}
+{{ toYaml .Values.extraInitContainers | indent 6 }}
+      {{- end }}
+      {{- end }}
+      containers:
+      - name: dashboards
+        securityContext:
+{{ toYaml .Values.securityContext | indent 10 }}
+        image: "{{ .Values.image }}:{{ .Values.imageTag }}"
+        imagePullPolicy: "{{ .Values.imagePullPolicy }}"
+        env:
+          {{- if .Values.opensearchURL }}
+          - name: OPENSEARCH_URL
+            value: "{{ .Values.opensearchURL }}"
+          {{- else if .Values.opensearchHosts }}
+          - name: OPENSEARCH_HOSTS
+            value: "{{ .Values.opensearchHosts }}"
+          {{- end }}
+          - name: SERVER_HOST
+            value: "{{ .Values.serverHost }}"
+            {{- if .Values.opensearchAccount.secret }}
+            - name: OPENSEARCH_USERNAME
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.opensearchAccount.secret }}
```

Comment:
Maybe rename this? `opensearchAccount.secret` is pretty vague. 

Perhaps: `opensearch.credentials.username` etc. etc.?

If this is a global secret, we perhaps should make the stuff that builds out this secret more identifiable by placing it in other places around the `values.yaml`

Reply:
👍 will make this change

Reply:
@DandyDeveloper I was thinking of keeping it the same way as we have in odfe helm [chart](https://github.com/opendistro-for-elasticsearch/opendistro-build/tree/main/helm#kibanayml-config)

Reply:
@TheAlgo Yeah that works. Just to be clear, you mean defining them as envs that the config will reference? 

The envs get populated via a secret? 

In any case, I think we should be creating that secret based on values input. I don't think it makes sense to expect a user to create an additional secret template, when Dashboards MUST have these credentials to get access to Elasticsearch?

Reply:
Yes I meant defining them as envs and then reference them. The envs will get filled by the secrets. I guess we can go ahead with the way it was done in open-distro. Once we get some feedback we can improve upon this.